donut/modules/editor/routes.py

```diff
@@ -0,0 +1,83 @@
+import flask
+import json
+import os
+import re
+
+from donut.modules.editor import blueprint, helpers
+from flask import current_app, redirect, url_for
+from donut.resources import Permissions
+from donut.auth_utils import check_permission
+
+
+@blueprint.route('/editor', methods=['GET', 'POST'])
+def editor():
+    '''
+    Returns the editor page where users can create and edit
+    existing pages
+    '''
+    input_text = "Hello world"
+    title = "TITLE"
+    inputt = flask.request.args.get('input_text')
+
+    if inputt != None:
+        input_text = helpers.read_markdown(inputt)
+        title = flask.request.args.get('title')
+
+    if check_permission(Permissions.ADMIN) and 'username' in flask.session:
+        return flask.render_template(
+            'editor_page.html', input_text=input_text, title=title)
+    else:
+        return flask.abort(403)
+
+
+@blueprint.route('/_change_title', methods=['POST'])
+def change_title():
+    '''
+    Allows one to change a file/page title.
+    '''
+    title = flask.request.form['title']
+    old_title = flask.request.form['old_title']
+    if check_permission(Permissions.ADMIN) and 'username' in flask.session:
+        helper.rename_title(old_title, title)
+
+
+@blueprint.route('/pages/_save', methods=['POST'])
+def save():
+    '''
+    Actually saves the data from the forms as markdown
+    '''
+    markdown = flask.request.form['markdown']
+    title = flask.request.form['title']
+    title_res = re.match("^[0-9a-zA-Z.\/_\- ]*$", title)
```

Comment:
You check the title multiple times, also in script.js:change_title(). What is the proper place to check (try to avoid redundant checks)?

Reply:
I'll change it to only check in the server then.

Reply:
also comment what this matches for future-proofing.